tests/lustre/typechecking/shouldPass/test_syntax.lus

```diff
@@ -0,0 +1,872 @@
+  (X, Y) = (E, F) -> pre (X, Y);
+  -- (X, Y) = (E, F) when C;
+  -- (X, Y) = N3(1, 2, 3);
+tel
+
+
+node NN (a, b: int) returns (c, d: int);
+let (c, d) = (a, b); tel
+
+-- node x62 (E, F, G, H: int) returns (A, B, C, D: int);
+-- let 
+--   (A, B, C, D) = (NN(E, F), NN(G, H));
+-- tel;
+
+/* TODO: 
+-- Node with structural assignments 
+node x63 (e: [bool, [int, int]]) returns (c: bool; a, b: int; A: [int, int]);
+let
+  [c, [a, b]] = e;
+  [c, A] = e;
+  a = e[1][0]; b = e[1][1];
+  A = e[1];
+  [c, [A[0], A[1]]] = e;
+tel;
+*/
+
+
+-- Section 7
+
+node N7 (x1, x2, x3: int) returns (y1, y2, y3: int);
+let 
+  (y1, y2, y3) = (x1, x2, x3);
+tel
+
+-- Node with a condact call 
+node x71 (C: bool; x1, x2, x3: int; v1, v2, v3: int) returns (y1, y2, y3: int);
+let 
+  (y1, y2, y3) = condact(C, N7(x1, x2, x3), v1, v2, v3);
+
+/* TODO:
+-- Equivalent expression to condact call
+  (y1, y2, y3) = if C then current(N7(x1, x2, x3) when C) 
+                 else (v1, v2, v3) -> pre(y1, y2, z3);
+*/
+tel
+
+-- node x72 (e1 : int; e2: real; e3: bool) returns (e: [int, real, bool]; f: real);
+-- let
+--   e = (e1, e2, e3);
+--   f = e[1];
+-- tel
+
+node x73 (a, b, c, e, f: int; d: bool) returns (e1, e2, e4: int; e3: bool);
+let
+  e1 = a + 2 * b;
+  e2 = a + b - c;
+  e3 = not (a = b);
+  e4 = if d then a else e -> f;
+tel
+
+node x74 (e1, e2, e3: int; f2: bool) returns (e: int^3; f: [int, bool, int]);
+let
+   e = [e1, e2, e3];
+tel
+
+/* TODO: 
+-- Node with array slices 
+node x75 (a: int^5) returns (d1: int; a23: int^2);
+const A = [1, 2, 3, 4, 5, 6];
+var
+  b1 : int;
+  b2, b3: int^3;
+  b4 : int^1;
+  b5: int; 
+let
+  d1 = a[0];
+  a23 = a[2..3];
+  b1 = A[1];
+  b2 = A[2..4];
+  b3 = A[4..2];
+  b4 = A[2..2];
+  b5 = A[2..4][0];
+tel
+*/
+
+type T = int;
+
+/* TODO: 
+-- Node with array slices and concatenations 
+node x76 (A: T^10^5) returns (B1: T^10^2; B2: T^2^4; B3: int^5 );
+let
+  B1 = A[1..4][1..2];
+  B2 = A[1..4, 1..2];
+  B3 = [1, 2, 3] | [4, 3];
+tel
+*/
+
+/* TODO: 
+-- Node using array slices and concatenations
+node EQ (const n: int; A, B: int^n) returns (equal: bool);
+var C, E: bool^n;
+let
+  equal = C[n-1];
+  C = [E[0]]|C[0..n-2] and E[1..n-1];
+  E = (A = B);
+tel
+
+-- Node using array slices and concatenations
+node DELAY (const n, default: int;X: int) returns (delayed: int);
+var window: int^n;
+let
+  delayed = window[n-1];
+  window = [X]|(default^(n-1) -> pre(window[0..n-2]));
+tel
+*/
+
+-- ------------------------------------------------------------
+-- Examples from the V6 draft manual
+-- ------------------------------------------------------------
+
+-- Enumerated types 
+type color1 = enum { blue, white, black };
+type color2 = enum { green, orange, yellow };
+
+-- Node with enumerated types 
+node enum0 (x: color1) returns (y: color2);
+let
+  y = if x = blue then green else if x = white then orange else yellow;
+tel
+
+type complex1 = { re: real; im: real };
+const j = complex1 { re = -(3.0)/2.0; im = 3.0/2.0 };
```

Comment:
This test fails. Error message seems to be wrong:

> line 797 col. 27: Failure cause: Type error:  Cannot apply arguments of type real and type real to obtain type int to operator of type real->real->real

Reply:
I am removing this file as it is a copy of [`syntax-test.lus`](https://github.com/kind2-mc/kind2/blob/develop/examples/syntax-test.lus). This error is now fixed that is now checked in `test_const_decls.lus`